### PR TITLE
Spotify App

### PR DESCRIPTION
**Pull request by jaumeserr**

### CODE REVIEW

File: staff/jaume-serradell/spotify-app/index.html [hunk 1]
```diff
@@ -0,0 +1,16 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <meta http-equiv="X-UA-Compatible" content="ie=edge">
+    <title>Document</title>
```
Comment (maiderhernandorena) [inline staff/jaume-serradell/spotify-app/index.html]: You should put a title on your files (on the head)

File: staff/jaume-serradell/spotify-app/scripts/main.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
+// my presentation logic
```
Comment (maiderhernandorena) [inline staff/jaume-serradell/spotify-app/scripts/main.js]: You should use 'use strict' at the begining

File: staff/jaume-serradell/spotify-app/scripts/main.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+// optional, reduce the size of the restaurants loaded in memory
+// restaurants.splice(100);
```
Comment (maiderhernandorena) [inline staff/jaume-serradell/spotify-app/scripts/main.js]: You should clean your code! In this file os Spotify App, these two coments are not necessary; you are not working with restaurants. : D

File: staff/jaume-serradell/spotify-app/scripts/my-components.js [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+ * @param {string} title The item title
+ * @param {string} info The information about an item
+ * @param {string} image The image of the item
+ */
+function DetailPanel(title, info, image, link, song) {
+    Panel.call(this, title, 'section');
+
+    var p = document.createElement('p');
+    p.innerText = info;
+
```
Comment (gbaste) [inline staff/jaume-serradell/spotify-app/scripts/my-components.js]: Please, clean the code, you need delete the spaces

File: staff/jaume-serradell/spotify-app/scripts/main.js (hunk 2 above)
Comment (ghost) [reply staff/jaume-serradell/spotify-app/scripts/main.js]: 'use stricti' mf

File: staff/jaume-serradell/spotify-app/scripts/main.js [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+var search = new SearchPanel();
+
+search.onSearch(function (query) {
+    logic.token = 'BQDvjaDo0bwTjgm_AuMTpolUiJXNApTsQ0qKK5HK9jcbYqn9MvAn1gMwlomsvC-th22arnA2LTgLaCL-9UlhGOJUD_5aSRZXDwrRPPRrtg1HrlXaNWN3BTwoL_cBlzcYXAc2VOSdz0FgsLFGZe8-JY4_5zZRlnf3Uz6LvBp_tAOzzwa6qq8z8IIflnnIHA_iAjN0lDOC9aC2WZ5nxC_bdV3aSjhkkbrq0D6whgvjcizWYdVMBeXX1rCiC3G-bXHunaaABhEmA40';
+    logic.searchArtists(query)
+        .then(function (artists) {
+            results.updateResults(artists.map(function (artist) {
```
Comment (ghost) [inline staff/jaume-serradell/spotify-app/scripts/main.js]: you should pass the entire array in order to manage it later (to include images and others).

File: staff/jaume-serradell/spotify-app/scripts/main.js [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+});
+
+var results = new ResultsList();
+
+var DEFAULT_IMAGE = 'https://i.pinimg.com/originals/37/2a/2d/372a2d5e8a32991bb19982271d0762fe.jpg';
+
+results.onItemClick(function (id) {
+    logic.retrieveAlbumsByArtistId(id)
+        .then(function (albums) {
+            resultsAlbum.updateResults(albums.map(function (album) {
```
Comment (ghost) [inline staff/jaume-serradell/spotify-app/scripts/main.js]: you should pass the entire array in order to manage it later (to include images and others).

File: staff/jaume-serradell/spotify-app/scripts/main.js [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+        });
+});
+
+var resultsAlbum = new ResultsList();
+
+
+resultsAlbum.onItemClick(function (id) {
+    logic.retrieveTracksByAlbumId(id)
+        .then(function (tracks) {
+            resultsAlbumTracks.updateResults(tracks.map(function (track) {
```
Comment (ghost) [inline staff/jaume-serradell/spotify-app/scripts/main.js]: you should pass the entire array in order to manage it later (to include images and others).

File: staff/jaume-serradell/spotify-app/scripts/main.js [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        })
+        .catch(function (error) {
+            alert('Sorry, we have temporary problem, try again later.');
+        });
+});
+
+var resultsAlbumTracks = new ResultsList();
+
+
+resultsAlbumTracks.onItemClick(function (id) {
```
Comment (ghost) [inline staff/jaume-serradell/spotify-app/scripts/main.js]: You should overwrite the updateResults() function in order to include more information.

File: staff/jaume-serradell/spotify-app/scripts/main.js [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+            }));
+
+            detailContainer.clear();
+        })
+        .catch(function (error) {
+            alert('Sorry, we have temporary problem, try again later.');
+        });
+});
+
+var resultsAlbum = new ResultsList();
```
Comment (ghost) [inline staff/jaume-serradell/spotify-app/scripts/main.js]: You should overwrite the updateResults() function in order to include more information.

File: staff/jaume-serradell/spotify-app/scripts/main.js [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+            }));
+
+            detailContainer.clear();
+        })
+        .catch(function (error) {
+            alert('Sorry, we have temporary problem, try again later.');
+        });
+});
+
+var results = new ResultsList();
```
Comment (ghost) [inline staff/jaume-serradell/spotify-app/scripts/main.js]: You should overwrite the updateResults() function in order to include more information.

**Review by ghost — COMMENTED** [review]
Some changes.

File: staff/jaume-serradell/spotify-app/index.html (hunk 1 above)
Comment (jaumeserr) [reply staff/jaume-serradell/spotify-app/index.html]: Done!

File: staff/jaume-serradell/arrays-native/scripts/includes.test.js [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    beforeEach(function () {
+        pets = ['cat', 'dog', 'bat'];
+    });
+
+    it('should include and operate correctly', function() {
+        //check if the value passed on the param is include in the array and return a boolean
+        var result = pets.includes('cat')
+        // expected output: true
+        
+        expect(result).toBe(true)
```
Comment (Javier1177) [inline staff/jaume-serradell/arrays-native/scripts/includes.test.js]: You missed semicolon

File: staff/jaume-serradell/arrays-native/scripts/includes.test.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
...
+        expect(pets.includes('mouse')).toBe(false)
```
Comment (Javier1177) [inline staff/jaume-serradell/arrays-native/scripts/includes.test.js]: You missed semicolon!

File: staff/jaume-serradell/spotify-app/scripts/main.js (hunk 2 above)
Comment (jaumeserr) [reply staff/jaume-serradell/spotify-app/scripts/main.js]: Done!

File: staff/jaume-serradell/spotify-app/scripts/main.js (hunk 3 above)
Comment (jaumeserr) [reply staff/jaume-serradell/spotify-app/scripts/main.js]: Done!

File: staff/jaume-serradell/arrays-native/scripts/indexof.test.js [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+describe('Array.prototype.indexof', function () {
+    var beasts;
+
+    beforeEach(function () {
+        beasts = ['ant', 'bison', 'camel', 'duck', 'bison'];
+    });
+
+    it('should include and operate correctly', function() {
+        //return the position of value passed by param
+        var result = beasts.indexOf('bison')
```
Comment (Javier1177) [inline staff/jaume-serradell/arrays-native/scripts/indexof.test.js]: You missed semicolon :(

File: staff/jaume-serradell/arrays-native/scripts/indexof.test.js [hunk 14]
```diff
@@ -0,0 +1,28 @@
...
+        // expected output: 1
+
+        //start from 'bison' count 2
+        var result2 = beasts.indexOf('bison', 2)
```
Comment (Javier1177) [inline staff/jaume-serradell/arrays-native/scripts/indexof.test.js]: You missed semicolon again :(

File: staff/jaume-serradell/arrays-native/scripts/indexof.test.js [hunk 15]
```diff
@@ -0,0 +1,28 @@
...
+        // expected output: 4
+
+        //'giraffe return -1 because it is not in the array
+        var result3 = beasts.indexOf('giraffe')
+        // expected output: -1
+        
+        expect(result).toBe(1)
```
Comment (Javier1177) [inline staff/jaume-serradell/arrays-native/scripts/indexof.test.js]: You missed semicolon again :(

File: staff/jaume-serradell/spotify-app/scripts/my-components.js (hunk 4 above)
Comment (jaumeserr) [reply staff/jaume-serradell/spotify-app/scripts/my-components.js]: Done!

File: staff/jaume-serradell/arrays-native/scripts/join.test.js [hunk 16]
```diff
@@ -0,0 +1,28 @@
...
+
+        //return the array in a string with all values together
+        var result2 = elements.join('');
+        //expected output: FireWindRain
+
+        //return the array in a string with all values separarated by hyphen
+        var result3 = elements.join('-');
+        // expected output: Fire-Wind-Rain
+        
+        expect(result).toBe('Fire,Wind,Rain')
```
Comment (Javier1177) [inline staff/jaume-serradell/arrays-native/scripts/join.test.js]: You missed semicolon :(

File: staff/jaume-serradell/arrays-native/scripts/shift.test.js [hunk 17]
```diff
@@ -0,0 +1,20 @@
...
+        array1 = [1, 2, 3];
+    });
+
+    it('should substract and operate correctly', function() {
+        //Substract the first value of the array
+        var result = array1.shift()
+        // expected output: 1
+        // array1 output: [2,3]
+        
+        expect(array1).toEqual([2,3])
```
Comment (Javier1177) [inline staff/jaume-serradell/arrays-native/scripts/shift.test.js]: You missed semicolon again :(

File: staff/jaume-serradell/arrays-native/scripts/includes.test.js (hunk 11 above)
Comment (jaumeserr) [reply staff/jaume-serradell/arrays-native/scripts/includes.test.js]: Done!

File: staff/jaume-serradell/arrays-native/scripts/includes.test.js (hunk 12 above)
Comment (jaumeserr) [reply staff/jaume-serradell/arrays-native/scripts/includes.test.js]: Done!

File: staff/jaume-serradell/arrays-native/scripts/indexof.test.js (hunk 13 above)
Comment (jaumeserr) [reply staff/jaume-serradell/arrays-native/scripts/indexof.test.js]: Done!

File: staff/jaume-serradell/arrays-native/scripts/indexof.test.js (hunk 14 above)
Comment (jaumeserr) [reply staff/jaume-serradell/arrays-native/scripts/indexof.test.js]: Done!

File: staff/jaume-serradell/arrays-native/scripts/indexof.test.js (hunk 15 above)
Comment (jaumeserr) [reply staff/jaume-serradell/arrays-native/scripts/indexof.test.js]: Done!

File: staff/jaume-serradell/arrays-native/scripts/join.test.js (hunk 16 above)
Comment (jaumeserr) [reply staff/jaume-serradell/arrays-native/scripts/join.test.js]: Done!

File: staff/jaume-serradell/arrays-native/scripts/shift.test.js (hunk 17 above)
Comment (jaumeserr) [reply staff/jaume-serradell/arrays-native/scripts/shift.test.js]: Done!

**Pull request merged**